lib/Util/ChannelTrait.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace MovingImage\Client\VMPro\Util;
+
+use MovingImage\Client\VMPro\Entity\Channel;
+
+/**
+ * Helper methods for manipulating Channels.
+ */
+trait ChannelTrait
+{
+    /**
+     * Configures the parent/child relationships between Channels.
+     *
+     * @param Channel[] $channels - any iterable collection of Channel
+     *
+     * @return array
+     */
+    private function setChannelRelations($channels)
+    {
+        $indexedChannels = [];
+
+        /** @var Channel $channel */
+        foreach ($channels as $channel) {
+            $indexedChannels[$channel->getId()] = $channel;
+        }
+
+        foreach ($indexedChannels as &$channel) {
+            $parentId = $channel->getParentId();
+            if ($parentId && array_key_exists($parentId, $indexedChannels)) {
+                /** @var Channel $parent */
+                $parent = &$indexedChannels[$parentId];
```

Comment:
Why did you put "&" here?
Objects and arrays transferred by reference (by default!).
According to PHP manual: "_Do not use return-by-reference to increase performance. The engine will automatically optimize this on its own. Only return references when you have a valid technical reason to do so._"

Reply:
Objects are passed by reference, but arrays are not.

Reply:
You are right and not. They are references, until you change something in the array.
I thought you did it because of performance.

Reply:
No, it's not because of performance. If I don't use a reference, I end up with multiple instances of the same Channel object.

Reply:
But it could be that this is a leftover from the first draft and may not be necessary any more. I will check again, you might be right.

Reply:
👍